### PR TITLE
♻️ Refactore NavigationPath

### PR DESCRIPTION
**Pull request by ahmed-yamany**

### CODE REVIEW

File: Tests/RoutingTests/NavigationPathManageableTests.swift [hunk 1]
```diff
@@ -24,56 +24,65 @@ final class NavigationPathManageableTests: XCTestCase {
...
     enum Route: ViewDisplayable {
-        case settings
+        case settings, profile
 
         @ViewBuilder
         var viewToDisplay: some View {
             switch self {
             case .settings:
                 MockSettingsView()
+                case .profile:
```
Comment (JamesSedlacek) [inline Tests/RoutingTests/NavigationPathManageableTests.swift:84]: alignment for these lines

File: Tests/RoutingTests/NavigationPathManageableTests.swift [hunk 2]
```diff
@@ -24,56 +24,65 @@ final class NavigationPathManageableTests: XCTestCase {
...
+    func testPopToSpecifiedDestination() {
+        router.push([.settings, .profile, .settings, .profile, .settings])
+        router.pop(to: .profile)
+        XCTAssertEqual(router.stack.count, 4)
+        XCTAssertEqual(router.stack, [.settings, .profile, .settings, .profile])
     }
 }
 
-fileprivate class MockNavigationPathManager: NavigationPathManageable {
+fileprivate class MockNavigationPathManager: NavigationStackManageable {
```
Comment (JamesSedlacek) [inline Tests/RoutingTests/NavigationPathManageableTests.swift:72]:
```suggestion
fileprivate class MockNavigationStackManager: NavigationStackManageable {
```

File: Tests/RoutingTests/NavigationPathManageableTests.swift (hunk 1 above)
Comment (ahmed-yamany) [reply Tests/RoutingTests/NavigationPathManageableTests.swift]: sorry i didn't notice this

File: Tests/RoutingTests/NavigationPathManageableTests.swift (hunk 2 above)
Comment (ahmed-yamany) [reply Tests/RoutingTests/NavigationPathManageableTests.swift]: i made the change, check it

**Review by JamesSedlacek — APPROVED** [review]

**Pull request merged**